sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -205,6 +205,7 @@ case class AdaptiveSparkPlanExec(
 
       // Run the final plan when there's no more unfinished stages.
       currentPhysicalPlan = applyPhysicalRules(result.newPlan, queryStageOptimizerRules)
+      executionId.foreach(onUpdatePlan)
```

Comment:
Why is the plan sent out before `isFinalPlan = true` ? Isn't it the final plan ?

Reply:
It doesn't matter. `onUpdatePlan` only look at `currentPhysicalPlan`, so we must call `onUpdatePlan` after setting `currentPhysicalPlan`.

Reply:
Does it matter for what's displayed on UI or anyone interested in sql update event ?

Reply:
what do you mean? Even if you move `isFinalPlan = true` before this line, nothing gets changed.

Reply:
The root nodes of `Physical Plan` in `physicalPlanDescription`s  are different: `AdaptiveSparkPlan(isFinalPlan=true)` vs `AdaptiveSparkPlan(isFinalPlan=false)`. So are root nodes of `sparkPlan`

Reply:
ah good catch! somehow I misread the code and thought it doesn't matter.

For UI it doesn't matter as we will strip the `AdaptiveSparkPlanExec`, but if there is a spark listener catching this event, it matters.

Can you send a PR to fix it? Thanks!

Reply:
Sure. Thanks for your reply.

Reply:
I've sent https://github.com/apache/spark/pull/26983. Please review.